cdap-master/src/main/java/co/cask/cdap/data/tools/ReplicationStatusTool.java

```diff
@@ -197,57 +242,76 @@ private static void checkHBaseReplicationComplete(Map<String, Long> masterTimeMa
     }
   }
 
-  static boolean checkDifferences(Set<String> masterRegions, Set<String> slaveRegions) {
-    boolean found = false;
-    Set<String> extraInMaster = Sets.difference(masterRegions, slaveRegions);
-    Set<String> extraInSlave = Sets.difference(slaveRegions, masterRegions);
+  static boolean checkDifferences(Set<String> masterMap, Set<String> slaveMap, String keyName) {
+    boolean different = false;
+    Set<String> extraInMaster = Sets.difference(masterMap, slaveMap);
+    Set<String> extraInSlave = Sets.difference(slaveMap, masterMap);
 
-    for (String regionName : extraInMaster) {
-      System.out.println("Region=" + regionName + " found on Master but not on Slave Cluster.");
-      found = true;
+    for (String key : extraInMaster) {
+      System.out.println(keyName + " " + key + " found on Master but not on Slave Cluster.");
+      different = true;
     }
 
-    for (String regionName : extraInSlave) {
-      System.out.println("Region=" + regionName + " found on Slave but not on Master Cluster.");
-      found = true;
+    for (String key : extraInSlave) {
+      System.out.println(keyName + " "  + key + " found on Slave but not on Master Cluster.");
+      different = true;
     }
-    return found;
+    return different;
   }
 
-  private static Map<String, Long> readMasterStatusFile(String masterStatusFile) throws IOException {
-    Map<String, Long> map = new HashMap<String, Long>();
+  private static void readMasterStatusFile(String masterStatusFile,
+                                           Map<String, String> hdfsMap,
+                                           Map<String, Long> hBaseMap) throws IOException {
     try (BufferedReader br = new BufferedReader(new FileReader(masterStatusFile))) {
       String line;
       //Read the shutdown Time
       masterShutdownTime = new Long(br.readLine());
-      //Read the checksum
-      masterChecksum = br.readLine();
 
-      //Read the rest into the region map
-      while ((line = br.readLine()) != null) {
+      //Read the checksums to the hdfs map
+      line = br.readLine();
+      while (!line.equals(checksumsEnd)) {
+        StringTokenizer st = new StringTokenizer(line, delimiter);
+        String file = st.nextToken();
+        String checksum = st.nextToken();
+        hdfsMap.put(file, checksum);
+        line = br.readLine();
+      }
+
+      //Read the table timestamps into the hbase map
+      line = br.readLine();
+      while (!line.equals(tablesEnd)) {
         StringTokenizer st = new StringTokenizer(line, delimiter);
         String region = st.nextToken();
         Long timestamp = new Long(st.nextToken());
-        map.put(region, timestamp);
+        hBaseMap.put(region, timestamp);
+        line = br.readLine();
       }
     }
-    return map;
   }
 
-  private static void writeMasterStatusFile(Map<String, Long> map, String filePath) throws IOException {
+  private static void writeMasterStatusFile(Map<String, Long> hBaseMap,
```

Comment:
for this method, rather than coming up with your own file format, it would be easier to create a class (for example, ReplicationState) that contains all the information you need, then write it out as json using the gson library. You would have a state class that has shutdownTime, regionState, and hdfsState, then use:
```
private static final Gson GSON = new Gson();
...
GSON.toJson(stateObj, bufferedWriter);
```
to write it out. Reading the file would be similarly straightforward:
```
stateObj = GSON.fromJson(fileReader, ReplicationState.class);
```

then, if we ever need to change what's in that file, it's just changing that ReplicationState class instead of adding reading and writing logic.